### PR TITLE
[#780] Support the RocketMQ TableSink based on the legacy Sink implementation

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

At present, RocketMQ connector supports the implementation of legacy SinkFunction interface, which needs the `DynamicTableSink` implementation to expand Table/SQL capabilities. RocketMQ connector should support the RocketMQ TableSink based on the legacy SinkFunction interface.

## Brief changelog

 - Introduce `RocketMQDynamicTableSink` based on the implementation `RocketMQSink` of legacy `SinkFunction` interface.

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by SteNicholas** [comment]
@vongosling @duhenglucky , could you help to review the RocketMQ TableSink based on the legacy Sink implementation?

**Comment by SteNicholas** [comment]
@vongosling @duhenglucky  , could you help to review the RocketMQ TableSink based on the legacy Sink implementation?

**Comment by ni-ze** [comment]
I can help make a review, but also need approve from write permission.

**Review by vongosling — APPROVED** [review]

File: rocketmq-flink/src/main/java/org/apache/rocketmq/flink/sink/table/RocketMQDynamicTableSink.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        this.writeKeysToBody = writeKeysToBody;
+        this.keyColumns = keyColumns;
+        this.metadataKeys = Collections.emptyList();
+    }
+
+    @Override
+    public ChangelogMode getChangelogMode(ChangelogMode requestedMode) {
+        ChangelogMode.Builder builder = ChangelogMode.newBuilder();
+        for (RowKind kind : requestedMode.getContainedKinds()) {
+            if (kind != RowKind.UPDATE_BEFORE) {
```
Comment (ni-ze) [inline rocketmq-flink/src/main/java/org/apache/rocketmq/flink/sink/table/RocketMQDynamicTableSink.java:109]: Why this RowKind.UPDATA_BEFORE kind of ChangelogMode is excluded？

File: rocketmq-flink/src/test/java/org/apache/rocketmq/flink/sink/table/RocketMQDynamicTableSinkFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.flink.table.api.DataTypes.STRING;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/** Tests for {@link org.apache.rocketmq.flink.sink.table.RocketMQDynamicTableSinkFactory}. */
+public class RocketMQDynamicTableSinkFactoryTest {
```
Comment (ni-ze) [inline rocketmq-flink/src/test/java/org/apache/rocketmq/flink/sink/table/RocketMQDynamicTableSinkFactoryTest.java:46]: It is look like only this test can not cover all change code.

**Review by ni-ze — COMMENTED** [review]
have been comment in the code.

**Review by duhenglucky — APPROVED** [review]

**Pull request merged**